controllers/kubeadmconfig_controller.go

```diff
@@ -95,6 +96,8 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 	machine, err := util.GetOwnerMachine(ctx, r.Client, config.ObjectMeta)
 	if err != nil {
 		log.Error(err, "could not get owner machine")
+                config.Status.ErrorStatus = CreateKubeadmconfigError
+                config.Status.ErrorReason = "could not get owner machine"
 		return ctrl.Result{}, err
```

Comment:
should this even be returning err?

Reply:
No, I don't think it should. We should wait for the CAPI machine controller to set the ownerReference in the KubeadmConfig to the Machine. Once that happens, this controller will get a natural update event and rereconcile the KubeadmConfig.

Reply:
Agreed, no need to return an error here.

Reply:
all set I think